native/src/main/java/io/ballerina/stdlib/task/listener/ListenerAction.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.task.listener;
+
+import io.ballerina.runtime.api.Environment;
+import io.ballerina.runtime.api.concurrent.StrandMetadata;
+import io.ballerina.runtime.api.creators.ValueCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.values.BDecimal;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BObject;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.stdlib.task.objects.TaskManager;
+import io.ballerina.stdlib.task.utils.ModuleUtils;
+import io.ballerina.stdlib.task.utils.Utils;
+
+import java.util.Map;
+
+import static io.ballerina.stdlib.task.utils.TaskConstants.JOB_ID;
+
+public class ListenerAction {
+    public static final String NATIVE_LISTENER_KEY = "TASK_NATIVE_LISTENER";
```

Comment:
Unless we are re-using these constants in other classes we should mark them as private. And if we are re-using them, can't we create a class as Constants and add them there.

Reply:
-1 for defining a ["`Constants`" class](https://www.baeldung.com/java-constants-good-practices#2-a-large-global-constants-class).

Reply:
@ThisaruGuruge The TaskConstants class is already merged. I have just reused a constant from it. Shall I remove the class?